attacks/reconstruction/reconstruction_module.py

```diff
@@ -115,55 +115,29 @@ def create_dicts(data, non_income_data, plausible_variable_combinations):
-                                                                    data_n = max(1, count.value)
-                                                                    )
+            mean = sn.dp_mean(priv_data, privacy_usage={'epsilon': 0.1})
             # get median
-            median = wn.dp_median(data = wn.cast(priv_data, 'FLOAT'),
-                                                                    privacy_usage = {'epsilon': 0.1},
-                                                                    data_lower = 0.,
-                                                                    data_upper = 100_000.,
-                                                                    data_n = max(1, count.value)
-                                                                    )
+            median = sn.dp_median(priv_data, privacy_usage={'epsilon': 0.1})
             # get min
-            _min = wn.dp_minimum(data = wn.cast(priv_data, 'FLOAT'),
-                                                                    privacy_usage = {'epsilon': 0.1},
-                                                                    data_lower = 0.,
-                                                                    data_upper = 100_000.,
-                                                                    data_n = max(1, count.value)
-                                                                    )
-
+            _min = sn.dp_minimum(priv_data, privacy_usage={'epsilon': 0.1})
             # get max
-            _max = wn.dp_maximum(data = wn.cast(priv_data, 'FLOAT'),
-                                                                    privacy_usage = {'epsilon': 0.1},
-                                                                    data_lower = 0.,
-                                                                    data_upper = 100_000.,
-                                                                    data_n = max(1, count.value)
-                                                                    )
-        analysis.release()
-        priv_mean_income_dict['__'.join(combination)] = min(max(0, mean.value), 100_000)
-        priv_median_income_dict['__'.join(combination)] = min(max(0, median.value), 100_000)
-        priv_min_income_dict['__'.join(combination)] = min(max(0, _min.value), 100_000)
-        priv_max_income_dict['__'.join(combination)] = min(max(0, _max.value), 100_000)
+            _max = sn.dp_maximum(priv_data, privacy_usage={'epsilon': 0.1})
+            priv_count_dict['__'.join(combination)] = max(0, count.value)
+            priv_mean_income_dict['__'.join(combination)] = min(max(0, mean.value), 100_000)
+            priv_median_income_dict['__'.join(combination)] = min(max(0, median.value), 100_000)
+            priv_min_income_dict['__'.join(combination)] = min(max(0, _min.value), 100_000)
+            priv_max_income_dict['__'.join(combination)] = min(max(0, _max.value), 100_000)
+
+            analysis.release()
```

Comment:
This release is unnecessary- the first time you call a .value that does not exist, `analysis.release()` is called internally. If you want a visual `.release()`, then move it to line 134. Minor pick, non-blocking.

Reply:
I have made the change you recommended. The analysis.release() is now before the join statements. It is confusing to me that the release is optional, thus I would imagine it might be confusing to others less familiar with smartnoise-core.